Libplanet/Blockchain/BlockChain.cs

```diff
@@ -149,6 +169,98 @@ public Block<T> Tip
             }
         }
 
+        /// <summary>
+        /// Gets the block corresponding to the <paramref name="hash"/>.
+        /// </summary>
+        /// <param name="hash">A <see cref="HashDigest{T}"/> of the <see cref="Block{T}"/> to get.
+        /// </param>
+        /// <exception cref="KeyNotFoundException">Thrown when there is no <see cref="Block{T}"/>
+        /// with a given <paramref name="hash"/>.</exception>
+        public Block<T> this[HashDigest<SHA256> hash]
+        {
+            get
+            {
+                _rwlock.EnterReadLock();
+                try
+                {
+                    return _blocks[hash];
+                }
+                finally
+                {
+                    _rwlock.ExitReadLock();
+                }
+            }
+        }
+
+        /// <summary>
+        /// Gets the transaction corresponding to the <paramref name="txId"/>.
+        /// </summary>
+        /// <param name="txId">A <see cref="TxId"/> of the <see cref="Transaction{T}"/> to get.
+        /// </param>
+        /// <exception cref="KeyNotFoundException">Thrown when there is no
+        /// <see cref="Transaction{T}"/> with a given <paramref name="txId"/>.</exception>
+        public Transaction<T> this[TxId txId]
```

Comment:
Instead of overloading this and making indexer able to return two distinct types, how about declare a method like `GetTransaction(TxId)`?

Reply:
That would be better. Fixed.